pxf/pxf-api/src/main/java/org/apache/hawq/pxf/api/utilities/Utilities.java

```diff
@@ -151,4 +157,78 @@ public static String maskNonPrintables(String input) {
         }
         return input.replaceAll("[^a-zA-Z0-9_:/-]", ".");
     }
+
+    /**
+     * Parses input data and returns fragment metadata.
+     * 
+     * @param inputData input data which has protocol information
+     * @return fragment metadata
+     * @throws IllegalArgumentException if fragment metadata information wasn't found in input data
+     * @throws Exception
+     */
+    public static FragmentMetadata parseFragmentMetadata(InputData inputData) throws Exception {
+        byte[] serializedLocation = inputData.getFragmentMetadata();
+        if (serializedLocation == null) {
+            throw new IllegalArgumentException("Missing fragment location information");
+        }
+        try (ByteArrayInputStream bytesStream = new ByteArrayInputStream(serializedLocation);
```

Comment:
since you're not using bytesStream directly, can collapse to 
ObjectInputStream objectStream = new ObjectInputStream( new ByteArrayInputStream(serializedLocation))